### PR TITLE
Fix Autobahn tests

### PR DESCRIPTION
**Pull request by joux3**
The test files used an older API for `message` events.

### CODE REVIEW

**Comment by lpinca** [comment]
Both files can be simplified a lot. This is what I used recently:

**Server**

```js
'use strict';

const WebSocket = require('ws');

const wss = new WebSocket.Server({
  perMessageDeflate: { threshold: 0 },
  port: 9001
}, () => console.log('server listening on *:9001'));

wss.on('connection', (ws) => {
  ws.on('error', (err) => console.error(err));
  ws.on('message', (msg) => ws.send(msg));
});
```

**Client**

```js
'use strict';

const WebSocket = require('ws');

let currentTest = 1;
let testCount;

function nextTest () {
  let ws;

  if (currentTest > testCount) {
    ws = new WebSocket('ws://localhost:3000/updateReports?&agent=new-parser');
    return;
  }

  console.log(`Running test case ${currentTest}/${testCount}`);

  ws = new WebSocket(`ws://localhost:9001/runCase?case=${currentTest}&agent=ws`);

  ws.on('error', (err) => console.error(err.stack));
  ws.on('message', (data) => ws.send(data));
  ws.on('close', () => {
    currentTest++;
    process.nextTick(nextTest);
  });
}

const ws = new WebSocket('ws://localhost:9001/getCaseCount');

ws.on('message', (data) => {
  testCount = +data;
});
ws.on('close', () => {
  if (testCount > 0) nextTest();
});
```

Since I totally forgot to do it, would you mind updating this PR to also modernize/clean up the code.

Thanks!

**Comment by 3rd-Eden** [comment]
@lpinca Yeah, thats why I didn't want to merge this yet. I knew you had it working against the current code base.

**Comment by joux3** [comment]
Done. Though my files look awfully like the ones you just posted.

File: test/autobahn-server.js [hunk 1]
```diff
@@ -1,31 +1,12 @@
...
-      process.exit();
-    });
-  } catch (e) {
-    process.exit();
-  }
-});
-
-var wss = new WebSocketServer({port: 8181});
-wss.on('connection', function (ws) {
+const wss = new WebSocket.Server({port: 8181});
```
Comment (lpinca) [inline test/autobahn-server.js]: By default Autobahn runs on port 9001, can you please use this port?

File: test/autobahn-server.js [hunk 2]
```diff
@@ -1,31 +1,12 @@
...
+wss.on('connection', (ws) => {
   console.log('new connection');
```
Comment (lpinca) [inline test/autobahn-server.js]: I would drop this line, it's just useless noise.

File: test/autobahn-server.js [hunk 3]
```diff
@@ -1,31 +1,12 @@
...
-  ws.on('message', function (data, flags) {
-    ws.send(flags.buffer, {binary: flags.binary === true});
-  });
-  ws.on('error', function () {
-    console.log('error', arguments);
+  ws.on('message', (data) => {
+    ws.send(data);
```
Comment (lpinca) [inline test/autobahn-server.js]: Can you merge this line with the one above?

File: test/autobahn-server.js (hunk 1 above)
Comment (joux3) [inline test/autobahn-server.js]: On the other hand the Autobahn report generator also runs on `9001`. Maybe I'll add a runtime option for that while defaulting to `9001`.

File: test/autobahn.js [hunk 4]
```diff
@@ -1,55 +1,34 @@
...
+  ws.on('message', (data) => ws.send(data));
+  ws.on('close', () => {
+    currentTest++;
     process.nextTick(nextTest);
   });
-  ws.on('error', function (e) {});
+  ws.on('error', (e) => console.error(e));
 }
 
 var ws = new WebSocket('ws://localhost:9001/getCaseCount');
```
Comment (lpinca) [inline test/autobahn.js]: Feel free to ignore: `const` instead of `var`

**Comment by lpinca** [comment]
Thanks a lot.

**Pull request merged**